kernel/solo5.h

```diff
@@ -102,15 +129,17 @@ int solo5_blk_rw(void);
  */
 
 /*
- * Writes (n) bytes from (*buf) to the console. Returns number of bytes
- * actually written.
+ * Writes (size) characters from (*buf) to the console.
+ *
+ * Always succeeds, however the implementation may truncate messages (e.g. due
```

Comment:
I'm with you that there's no way to recover from most `write` errors.  I'd rephrase `Always succeeds` to reflect the actual behaviour  (although the virtio backend using `outb` can never fail AFAICS).  Do you expect that compilations always have assertions enabled (and leading to fatal errors)?  Or should there be a call to `exit()` instead of `assert` (sorry if I missed some note which explains how to use assertions in Solo5)?